src/platform/packages/shared/kbn-ftr-common-functional-services/services/es.ts

```diff
@@ -25,4 +33,116 @@ export function EsProvider({ getService }: FtrProviderContext): Client {
           authOverride: systemIndicesSuperuser,
         }
   );
+
+  const refreshInterval = config.get('indexRefreshInterval');
+  // do not optimize for cloud runs or if explicitly disabled
+  if (!!process.env.TEST_CLOUD || refreshInterval === false) {
+    return client;
+  }
+
+  const idxPatterns = [
+    '.kibana*',
+    '.internal*',
+    'logs*',
+    'metrics*',
+    'traces*',
+    'filebeat*',
+    'metricbeat*',
+  ];
+
+  function wrap<T, U extends keyof T>(obj: T, prop: U, cb: (m: T[U]) => T[U]) {
+    const original = obj[prop];
+    obj[prop] = cb(original);
+  }
+
+  await client.cluster.putComponentTemplate({
+    name: 'fast_refresh',
+    template: {
+      settings: {
+        refresh_interval: refreshInterval,
+      },
+    },
+  });
+
+  async function putFastRefreshSettings() {
+    await client.indices.putSettings({
+      index: idxPatterns,
+      allow_no_indices: true,
+      expand_wildcards: ['all'],
+      settings: {
+        index: {
+          refresh_interval: refreshInterval,
+        },
+      },
+    });
+  }
+
+  const { index_templates: idxTemplates } = await client.indices.getIndexTemplate({
+    name: '*',
+  });
+
+  const limiter = pLimit(10);
+
+  await Promise.all(
+    idxTemplates.map(async (tpl) => {
+      const next = {
+        name: tpl.name,
+        ...omit(tpl.index_template, 'created_date_millis', 'modified_date_millis', 'version'),
+        ignore_missing_component_templates: castArray(
+          tpl.index_template.ignore_missing_component_templates ?? []
+        ),
+        composed_of: uniq([...tpl.index_template.composed_of, 'fast_refresh']),
+      };
+
+      // there are some templates where the priority is higher than ES' parser supports
+      if (next.priority && next.priority >= Number.MAX_SAFE_INTEGER) {
+        return;
+      }
+
+      await limiter(() => client.indices.putIndexTemplate(next));
+    })
+  );
+
+  function wrapTemplate(tpl: IndicesPutIndexTemplateIndexTemplateMapping = {}) {
```

Comment:
As this is used only un the wrapper below it could be defined there as well.

nit: Usually the best practice is using arrow functions inside a top level function to avoid any potential issues with the context.